src/modules/AddOnProduct.rb

```diff
@@ -145,7 +145,9 @@ def main
         "sle-hae"       => [ "sle-ha" ],
         "SUSE_SLES_SAP" => [ "SLES_SAP" ],
         # SMT is now integrated into the base SLES
-        "sle-smt"       => [ "SLES" ]
+        "sle-smt"       => [ "SLES" ],
+        # SLES_SAP includes SLES. So SLES is not needed anymore
+        "SLES"          => [ "SLES_SAP" ]
```

Comment:
Um, this looks like a very interesting idea, but unfortunately this will probably not work correctly as it has some side effects.

This mapping is used when evaluating product upgrades, see https://github.com/yast/yast-packager/blob/d3c1f7e566612e507dc835f36dce879541bd44ec/src/modules/Packages.rb#L2655

In this case the SLES12 -> SLES4SAP12 upgrade would be considered OK, while originally it would report an error (a product removal) and a new product installation. And the upgrade would be blocked until user manually approved this (by manually selecting the product to install). With this change it would be considered OK.

IMHO every product change should be reported as a error to avoid changing the system by mistake (e.g. a typo in the URL). Although SLES4SAP12 is compatible with SLES12 I still think the user should explicitly confirm the upgrade to avoid surprises later.